nomad/structs/job.go

```diff
@@ -22,3 +22,41 @@ type JobServiceRegistrationsResponse struct {
 	Services []*ServiceRegistration
 	QueryMeta
 }
+
+// RequiredNativeServiceDiscovery identifies which task groups, if any, within
+// the job are utilising Nomad native service discovery.
+func (j *Job) RequiredNativeServiceDiscovery() map[string]bool {
+	groups := make(map[string]bool)
+
+	for _, tg := range j.TaskGroups {
+
+		// It is possible for services using the Nomad provider to be
+		// configured at the task group level, so check here first.
+		if requiresNativeServiceDiscovery(tg.Services) {
+			groups[tg.Name] = true
+			continue
+		}
+
+		// Iterate the tasks within the task group to check the services
+		// configured at this more traditional level.
+		for _, task := range tg.Tasks {
+			if requiresNativeServiceDiscovery(task.Services) {
+				groups[tg.Name] = true
+				continue
+			}
+		}
+	}
+
+	return groups
+}
+
+// requiresNativeServiceDiscovery identifies whether any of the services passed
+// to the function are utilising Nomad native service discovery.
+func requiresNativeServiceDiscovery(services []*Service) bool {
+	for _, tgService := range services {
+		if tgService.Provider == ProviderNomad {
```

Comment:
Oops didn't notice the name `ProviderNomad` before. Should probably rename that to `ServiceProviderNomad` since there will likely be other kinds of providers added someday.